tests/test_dagviz.py

```diff
@@ -0,0 +1,100 @@
+'''
+Tests for dagviz and FlowImage class.
+'''
+
+import pytest
+import io
+import warnings
+import unittest.mock as mock
+import xml
+from xml.etree import ElementTree as ET
+import pydot
+import networkx as nx
+import PIL
+import bionic.dagviz as dagviz
+
+
+@pytest.fixture
+def pydot_graph():
+    '''Create empty pydot graph for testing'''
+    return pydot.Dot(graph_type='graph')
+
+
+@pytest.fixture
+def flow_image(pydot_graph):
+    '''Create FlowImage fixture for testing'''
+    return dagviz.FlowImage(pydot_graph)
+
+
+def test_flowimage_typing(flow_image):
+    '''Check types for attributes of flowImage'''
+    display_str = flow_image._repr_svg_()
+    try:
+        ET.fromstring(flow_image.xml_text)
+    except xml.etree.ElementTree.ParseError:
+        pytest.fail("FlowImage.xml_text not well formed XML")
+    assert isinstance(flow_image, dagviz.FlowImage)
+    assert isinstance(flow_image.pil_image, PIL.Image.Image)
+    assert isinstance(flow_image.xml_text, bytes)
+    assert isinstance(display_str, str)
+
+
+@pytest.mark.parametrize("fp_input",
+                         ['flow.svg', '/path/to/folder/flow.svg', 'filename.svg']
+                         )
+def test_save_flowimage_svg(fp_input, flow_image):
+    '''Test custom saving for SVG files'''
+    with mock.patch('builtins.open') as mock_open:
+        flow_image.save(fp_input)
+        assert mock_open.call_count == 1
+        mock_open.assert_called_with(fp_input, 'wb')
+
+
+def test_save_flowimage_warning(flow_image):
+    '''Warning is thrown if svg image is saved with format options (not compatible'''
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter('always')
+        flow_image.save('flow.svg', format='formatting options')
+        assert len(w) == 1
+
+
+@pytest.mark.parametrize("fp_input",
+                         ['flow.png',
+                          '/path/to/folder/flow.png',
+                          'filename.png',
+                          'machine_learning.jpeg'])
+@mock.patch.object(PIL.Image.Image, 'save')
```

Comment:
The use of mocks sounds like a great idea for testing `show`. For `save`, though, wouldn't it be a similar level of effort to actually write to a file or buffer and test that it actually contains a PNG/SVG? This would additionally test that (a) we passed the right arguments to `save` and (b) that `save` actually does what the user expects.